core/src/main/scala/kafka/log/Log.scala

```diff
@@ -615,10 +615,7 @@ class Log(val dir: File,
         None
     }
 
-    targetSeg match {
-      case Some(segment) => segment.findOffsetByTimestamp(targetTimestamp)
-      case None => None
-    }
+    targetSeg.map(_.findOffsetByTimestamp(targetTimestamp).get)
```

Comment:
Sorry, I should have said `flatMap` instead of `map` in my original comment. If you do that, you don't need the `get` any more.